### DISCUSSION TITLE
how to stop/halt autogenstudio correctly when the the execution is "hang" ( like endless loops ) etc ?

### DISCUSSION THREAD

**Discussion by mohdfaaf** [microsoft/autogen#1691] [Q&A]
Hi 

I am new with autogenstudio... managed to install it and run in.. but for some tasks.. it took longer time.. so my question is 

how to stop/halt autogenstudio correctly when the the execution is "hang" ( like endless loops ) etc ? 

Thanks

**Comment by raslab** [comment]
The same here. I have connected ollama to autogen studio agents, but sometimes (I didn't found why) ollama with some models for some requests just stucks for 20+ minutes to generate random text... 
And right now I can't find any solution to stop autogen in this case. Even can't terminate service in terminal. Only when ollama service have restarted - this nightmare stops.

**Comment by jonmadison-amzn** [comment]
Currently, there's no way to do this; you'll have to do a `kill -9` based on the process id.

something like

```
ps -alx | grep "autogenstudio ui" | grep -v grep |  awk '{print $2}' |  xargs kill -9
```

i too am looking for something in the UI itself to stop execution; it's a pain.

i'd also like console output available in the UI, but the UI's pretty basic, in the sense that once a task is started, the rest of the UI is essentially blocked

**Comment by vanetreg** [comment]
This should be fixed ASAP because it just keeps burning money...

**Comment by medecau** [comment]
i'm not familiar with studio but could [`MaxMessageTermination`](https://microsoft.github.io/autogen/stable/reference/python/autogen_agentchat.conditions.html#autogen_agentchat.conditions.MaxMessageTermination) help break out of these loops?
